clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7708,7 +7708,8 @@ void Sema::ProcessDeclAttributeList(Scope *S, Decl *D,
   // good to have a way to specify "these attributes must appear as a group",
   // for these. Additionally, it would be good to have a way to specify "these
   // attribute must never appear as a group" for attributes like cold and hot.
-  if (!D->hasAttr<OpenCLKernelAttr>()) {
+  if (!(D->hasAttr<OpenCLKernelAttr>() ||
+        LangOpts.SYCLIsDevice || LangOpts.SYCLIsHost)) {
```

Comment:
LangOpts.SYCLIsHost variables name suggests that it's true for the host side of the SYCL application.
Could you clarify what should happen with this attribute on the host side? How will handle it?

Reply:
> LangOpts.SYCLIsHost variables name suggests that it's true for the host side of the SYCL application.

Correct.

> Could you clarify what should happen with this attribute on the host side? How will handle it?

The reason I did this was to avoid a diagnostic about unsupported attributes for host compilation. Since I didn't do anything to drop the attribute, it is probably emitted to LLVM IR (and ignored by optimizations & codegen) for host code.

Reply:
This is actually re-using of solution developed in intel/llvm#296

Reply:
AFAIK this attribute cannot be ignored, so we need somehow handle it when executing on host device.